### PR TITLE
Add ability to change included tags

### LINKED ISSUE #22345: Adding Test suites/collections/profiles to :dev
### Description

What would make the amazing `dev` experience even nicer would be the possibility to add different test "profiles" or "suites" that can be triggered via keystroke, besides the `r` for (re-)running the tests.

Example: Run my unit tests (that I've defined with `.*UnitTest`) with pressing `t` and `1`, run my UI tests (defined with `.*UITest`) with `t` + `2`, and so on.

Right now, I am running the `dev` mode in different versions of `quarkus.test.{exclude,include}-pattern`, but that requires a restart every time, I'm in a different "testing mode".

How we could do that is to define the different suites in the build Quarkus plugin section, or via property, such as `quarkus.test.suite[index].{exclude-include}-pattern` (`suite` only being a suggestion).

WDYT?

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by stuartwdouglas**
Fixes #22345

### CODE REVIEW

**Review by geoand — APPROVED** [review]

**Comment by famod** [comment]
@geoand can we get this in for 2.7? There was positive user feedback here: https://github.com/quarkusio/quarkus/issues/22345#issuecomment-999346861
The second part of that comment should be filed as a separate improvement issue, IMO.

I'll rebase this in the meantime...

**Review by geoand — APPROVED** [review]

File: core/deployment/src/main/java/io/quarkus/deployment/dev/testing/TestTracingProcessor.java [hunk 1]
```diff
@@ -146,4 +184,177 @@ public void visitCode() {
...
+            PatternCommand.class })
+    public static class TestCommand implements Command {
+
+        @Override
+        public CommandResult execute(CommandInvocation commandInvocation) throws CommandException, InterruptedException {
+            return CommandResult.SUCCESS;
+        }
+    }
+
+    @GroupCommandDefinition(name = "tags", description = "Test Commands", groupCommands = { IncludeTagsCommand.class,
```
Comment (famod) [inline core/deployment/src/main/java/io/quarkus/deployment/dev/testing/TestTracingProcessor.java:203]: `"Test Commands"` looks like a c&p mistake.
```suggestion
    @GroupCommandDefinition(name = "tags", description = "Tag Commands", groupCommands = { IncludeTagsCommand.class,
```
@geoand WDYT?

Comment (geoand) [reply core/deployment/src/main/java/io/quarkus/deployment/dev/testing/TestTracingProcessor.java:203]: Makes sense

Comment (famod) [reply core/deployment/src/main/java/io/quarkus/deployment/dev/testing/TestTracingProcessor.java:203]: I'll change it after CI is done.

Comment (famod) [reply core/deployment/src/main/java/io/quarkus/deployment/dev/testing/TestTracingProcessor.java]: Done.

**Comment by famod** [comment]
`KafkaConnectorTest` failure doesn't look related:
```
2022-01-10T00:57:44.4037235Z 2022-01-10 00:57:44,256 ERROR [🐳 .io/.11.2]] (build-74) Log output from the failed container:
2022-01-10T00:57:44.4038182Z sh: 1: /var/lib/redpanda/redpanda.sh: Text file busy
```

**Pull request merged**